pageserver/src/tenant/storage_layer/delta_layer.rs

```diff
@@ -669,7 +446,7 @@ impl DeltaLayerWriterInner {
     ///
     /// Finish writing the delta layer.
     ///
-    fn finish(self, key_end: Key) -> anyhow::Result<DeltaLayer> {
+    fn finish(self, key_end: Key, timeline: &Arc<Timeline>) -> anyhow::Result<ResidentLayer> {
```

Comment:
There's two layers of DeltaLayerWriter to enable what could be just `Option<DeltaLayerWriterInner>` ... I did not want to touch this, because me cleaning that up would just add more diffstat. With this I can just side-step all that..